source/opt/strength_reduction_pass.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "strength_reduction_pass.h"
+
+#include <algorithm>
+#include <cstdio>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "def_use_manager.h"
+#include "log.h"
+#include "reflect.h"
+
+namespace {
+// Utility function to count the number of trailing zeros in constVal.
```

Comment:
"Utility function" phrase isn't really helping here. Just say what it does.

Why not just name the function CountTrailingZeroes?  That describes what it does rather than what it could be used for.

Reply:
... after reading the body ... :   You need to say that constVal is assumed to be a power of 2.